persistent-test/src/RawSqlTest.hs

```diff
@@ -10,93 +13,126 @@ import PersistentTestModels
+                             ]
+        ret1 <- rawSql query []
+        ret2 <- rawSql query [] :: MonadIO m => SqlPersistT m [Entity (ReverseFieldOrder Person)]
+        liftIO $ ret1 @?= [Entity p1k p1]
+        liftIO $ ret2 @?= [Entity (RFOKey $ unPersonKey p1k) (RFO p1)]
+
+    it "permits prefixes" $ runDb $ do
+        let r1 = Relationship "Foo" Nothing
+        r1k <- insert r1
+        let r2 = Relationship "Bar" (Just r1k)
+        r2k <- insert r2
+        let r3 = Relationship "Lmao" (Just r1k)
+        r3k <- insert r3
+        let r4 = Relationship "Boring" (Just r2k)
+        r4k <- insert r4
+        escape <- getEscape
+        let query = T.concat
+                [ "SELECT ??, ?? "
+                , "FROM ", escape "Relationship", " AS parent "
+                , "LEFT OUTER JOIN ", escape "Relationship", " AS child "
+                , "ON parent.id = child.parent"
+                ]
+
+        result :: [(EntityWithPrefix "parent" Relationship, Maybe (EntityWithPrefix "child" Relationship))] <-
+            rawSql query []
+
+        liftIO $
+            coerce result `shouldMatchList`
+                [ (Entity r1k r1, Just (Entity r2k r2))
+                , (Entity r1k r1, Just (Entity r3k r3))
+                , (Entity r2k r2, Just (Entity r4k r4))
+                , (Entity r3k r3, Nothing)
+                , (Entity r4k r4, Nothing)
+                ]
+
+
+    it "OUTER JOIN" $ runDb $ do
+        let insert' :: (PersistStore backend, PersistEntity val, PersistEntityBackend val ~ BaseBackend backend, MonadIO m)
+                    => val -> ReaderT backend m (Key val, val)
+            insert' v = insert v >>= \k -> return (k, v)
+        (p1k, p1) <- insert' $ Person "Mathias"   23 Nothing
+        (p2k, p2) <- insert' $ Person "Norbert"   44 Nothing
+        (a1k, a1) <- insert' $ Pet p1k "Rodolfo" Cat
+        (a2k, a2) <- insert' $ Pet p1k "Zeno"    Cat
+        escape <- getEscape
+        let query = T.concat [ "SELECT ??, ?? "
+                             , "FROM ", person
+                             , "LEFT OUTER JOIN ", pet
+                             , " ON ", person, ".", escape "id"
+                             , " = ", pet, ".", escape "ownerId"
+                             , " ORDER BY ", person, ".", escape "name"]
+            person = escape "Person"
+            pet    = escape "Pet"
+        ret <- rawSql query []
+        liftIO $ ret @?= [ (Entity p1k p1, Just (Entity a1k a1))
+                         , (Entity p1k p1, Just (Entity a2k a2))
+                         , (Entity p2k p2, Nothing) ]
+
+    it "handles lower casing" $
+        runDb $ do
+            C.runConduitRes $ rawQuery "SELECT full_name from lower_case_table WHERE my_id=5" [] C..| CL.sinkNull
+            C.runConduitRes $ rawQuery "SELECT something_else from ref_table WHERE id=4" [] C..| CL.sinkNull
+
+    it "commit/rollback" $ do
+        caseCommitRollback runDb
+        runDb cleanDB
```

Comment:
this was a mix of 2 space and 4 space indent :| Fixed it here, but it gives a noisy diff.